include/unifex/task.hpp

```diff
@@ -172,20 +173,21 @@ struct _promise {
     decltype(auto) await_transform(Value&& value)
         // TODO fix this noexcept
         noexcept(is_nothrow_callable_v<tag_t<unifex::await_transform>, type&, Value>) {
-      if constexpr (derived_from<remove_cvref_t<Value>, _task_base>) {
-        // We are co_await-ing a unifex::task. Assume that the task transitions back to
-        // the correct context automatically and doesn't need an additional transition.
+      if constexpr (derived_from<remove_cvref_t<Value>, _task_base> ||
+          blocking_kind::always_inline == cblocking<Value>()) {
+        // We are co_await-ing a unifex::task or something that completes inline. We don't
+        // need an additional transition.
         return unifex::await_transform(*this, (Value&&) value);
       } else if constexpr (tag_invocable<tag_t<unifex::await_transform>, type&, Value>
           || detail::_awaitable<Value>) {
         // Either await_transform has been customized or Value is an awaitable. Either
         // way, we can dispatch to the await_transform CPO, then insert a transition back
         // to the correct execution context.
```

Comment:
Is it possible for the result of a customized `await_transform` to communicate that it completes inline?

Reply:
I suppose so. I should account for that.